chain/ethereum/src/chain.rs

```diff
@@ -300,20 +302,47 @@ impl Blockchain for Chain {
                 self.name, requirements
             ));
 
-        if self.firehose_endpoints.len() > 0 {
-            self.new_firehose_block_stream(deployment, start_blocks, adapter, filter)
-                .await
-        } else {
-            self.new_polling_block_stream(
-                deployment,
-                start_blocks,
-                adapter,
-                filter,
-                metrics,
-                unified_api_version,
+        self.new_firehose_block_stream(
```

Comment:
Seems like we don't need a separate two fns with the same name so this can be inlined here.

Reply:
done